test: automatically download brokerpaks for upgrade tests

blgm opened the pull request:
Setting up upgrade tests is painful and error prone as one has to manually download a brokerpak, it's associated ".envrc" file and corresponding broker version. This has now been automated, so it's much easier to run upgrade tests.

zucchinidev commented on acceptance-tests/upgrade/upgrade_suite_test.go:
```diff
@@ -1,24 +1,59 @@
 )
 
 func init() {
-	flag.StringVar(&releasedBuildDir, "releasedBuildDir", "../../../aws-released", "location of released version of built broker and brokerpak")
```
praise: thanks for remove this configuration

zucchinidev commented on acceptance-tests/upgrade/.gitignore:
```diff
@@ -0,0 +1 @@
+versions/
```
thoughts: we may want to add a `.gitkeep` file inside of the folder and ignore the rest

blgm commented:
@zucchinidev I have made some changes. Mainly:
- added a script to allow the brokerpaks to be listed
- added a script to allow a brokerpak to be prepared without running a test (meant to be used in CI)
- as a consequence, I removed ginkgo/gomega aspects as these only work in tests
- I did not update the `.gitignore` file as while your suggestion achieved the same thing, I couldn't see an advantage to the alternative suggestion. But I'm open to being persuaded that the alternative is a better approach.
